addons/l10n_br/models/account_move.py

```diff
@@ -0,0 +1,39 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models, fields
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_br_invoice_serial = fields.Char(related="journal_id.l10n_br_invoice_serial", readonly=True)
+
+    def _compute_l10n_latam_document_type(self):
+        """ Override for debit notes. This sets the same document type as the one on the origin. Cannot
+         override the defaults in the account.move.debit wizard because l10n_latam_invoice_document explicitly
+         calls _compute_l10n_latam_document_type() after the debit note is created. """
+        br_debit_notes = self.filtered(lambda m: m.state == "draft" and m.country_code == "BR" and m.debit_origin_id.l10n_latam_document_type_id)
+        for move in br_debit_notes:
+            move.l10n_latam_document_type_id = move.debit_origin_id.l10n_latam_document_type_id
```

Comment:
@vbe-odoo should we copy the document type from invoice to credit note as well? I think not right, because it will definitely be different?

Reply:
@jorenvo copying the Doc Type from the INV to the Credit Note is what we did on the Credit Note wizard when creating the Credit Note. As the Credit Note DOC type is the same as the one coming from the INV.

**For example**:
* Invoice with DOC Type NF-e (55) -> operation type "stadardSales"
* Credit Note DOC Type NF-e (55) -> operation type "salesRedund"

Is that what you are referring to? or something different?
Furthermore, for CredIt Notes in BR the user needs to unselect the boolean on the Journal that specifies that the sequence is independent from the invoice sequence:
![image](https://github.com/odoo/odoo/assets/45669815/1911083e-20d9-48ca-8550-69302c6818a6)

Reply:
Or we override the domain a bit harder, so it does not play a role.

Reply:
You could check if reversed_entry_id is set like you do with debit notes.  The debit note and credit note is a bit handled differently as for credit notes you can cancel out and then it generates an already posted credit note.  (which seemed less obvious to alter the sequence or doc type.  But for this, we could do it in the same way.  (as it simulates creating the credit note for the default doc type in the wizard)

Reply:
@jco-odoo 
What is functionally needed is that the Credit Note and Debit Note sequences follow the same sequence as the invoice DOC Type within a specific Journal.

Assuming the following Documents happen one after the other:
* Invoice NF-e (55) Number 100
* Credit Note NF-e (55) Number 101
* Debit Note NF-e (55) Number 102
* Invoice NF-e (55) Number 103
* Invoice NFC-e (65) Number 250 (this doc type has a different sequence)
* Credit Note NFC-e (65) Number 251
* Debit Note NFC-e (65) Number 252
* Invoice NFC-e (65) Number 253
* ....

Reply:
@vbe-odoo So you confirm that the credit note should normally have the same doc type as its original invoice?

Reply:
Yes, they do, and they share sequences. So setting the Doc Type from the Origin Inv as the default one on the wizard when creating the Credit Note will be useful, to avoid errors.
What it changes is the "operationType", and for this field we re hardcoding it depending on the transaction.

Reply:
In https://github.com/odoo/odoo/blob/cf03174af6001b24f6293ed30c095baa05b185c8/addons/l10n_latam_invoice_document/wizards/account_move_reversal.py#L60  we might just add the reversed_entry_id ...  (it is taking us further than needed maybe)

Reply:
Then we could make the compute for BR (override) let it put the reversed_entry.  But I suggest to not go that far for now.